internal/commands/user/delete/inputs.go

```diff
@@ -0,0 +1,193 @@
+
+	"github.com/AlecAivazis/survey/v2"
+)
+
+// input field names, per survey
+const (
+	inputFieldUser      = "user"
+	inputFieldProviders = "provider"
+	inputFieldState     = "state"
+	inputFieldStatus    = "status"
+)
+
+type inputs struct {
+	cli.ProjectAppInputs
+	State             realm.UserState
+	Users             []string
+	ProviderTypes     []string
+	Status            statusType
+	InteractiveFilter bool
+}
+
+func (i *inputs) Resolve(profile *cli.Profile, ui terminal.UI) error {
+	if err := i.ProjectAppInputs.Resolve(ui, profile.WorkingDirectory); err != nil {
+		return err
+	}
+
+	var err error
+	// Interactive set Status
+	if i.InteractiveFilter {
+		allUserStatuses := []string{statusTypeConfirmed.String(), statusTypePending.String()}
+		selectedStatuses := []string{}
+		defaultStatuses := []string{}
+		if i.Status != statusTypeNil {
+			newStatusType := statusType(i.Status)
+			err = newStatusType.Set(i.Status.String())
+			if err != nil {
+				return err
+			}
+			defaultStatuses = append(defaultStatuses, i.Status.String())
+		}
+		err := ui.AskOne(
+			&selectedStatuses,
+			&survey.MultiSelect{
+				Message: "Which user state would you like to filter confirmed users by? Selecting none is equivalent to selecting all.",
+				Options: allUserStatuses,
+				Default: defaultStatuses,
+			},
+		)
+		if err != nil {
+			return err
+		}
+		if len(selectedStatuses) == 1 {
+			i.Status = statusType(selectedStatuses[0])
+		} else {
+			i.Status = statusTypeNil
+		}
+	}
+	if i.Status == statusTypeConfirmed || i.Status == statusTypeNil {
+		// Interactive set Providers
+		if i.InteractiveFilter {
+			err := ui.AskOne(
+				&i.ProviderTypes,
+				&survey.MultiSelect{
+					Message: "Which provider(s) would you like to filter confirmed users by? Selecting none is equivalent to selecting all.",
+					Options: shared.ValidProviderTypes,
+					Default: i.ProviderTypes,
+				},
+			)
+			if err != nil {
+				return err
+			}
+		}
+
+		// Interactive set State
+		if i.InteractiveFilter {
+			allUserStates := []string{realm.UserStateEnabled.String(), realm.UserStateDisabled.String()}
+			selectedStates := []string{}
+			defaultStates := []string{}
+			if i.State != realm.UserStateNil {
+				newStateType := realm.UserState(i.State)
+				err = newStateType.Set(i.State.String())
+				if err != nil {
+					return err
+				}
+				defaultStates = append(defaultStates, i.State.String())
+			}
+			err := ui.AskOne(
+				&selectedStates,
+				&survey.MultiSelect{
+					Message: "Which user state would you like to filter confirmed users by? Selecting none is equivalent to selecting all.",
+					Options: allUserStates,
+					Default: defaultStates,
+				},
+			)
+			if err != nil {
+				return err
+			}
+			if len(selectedStates) == 1 {
+				i.State = realm.UserState(selectedStates[0])
+			} else {
+				i.State = realm.UserStateNil
+			}
+		}
+	}
+	return nil
+}
+
+func (i *inputs) ResolveUsers(ui terminal.UI, client realm.Client, app realm.App) error {
```

Comment:
Since user revoke needs this as well, it might be a good idea to move this function into a common place and not have it be a receiver to inputs. I assume that the return could then be changed to ([]string, error) where we return a list of userID strings